components/AposArea.astro

```diff
@@ -19,11 +19,34 @@ if (isEdit) {
   };
 }
 const Wrapper = isEdit ? 'div' : Fragment;
+const widgetOptions = getWidgetOptions(area.options);
+
+function getWidgetOptions(options) {
+  let widgets = options.widgets || {};
```

Comment:
I lifted this logic verbatim from apostrophe core for my piece of mind. Later we might refactor so that the back end provides this to us in a pre-baked form. We have some cleanup coming.